sdk/storage/azblob/zc_blob_client.go

```diff
@@ -139,91 +161,91 @@ func (b *BlobClient) Undelete(ctx context.Context) (BlobUndeleteResponse, error)
 // bandwidth of the blob. A block blob's tier determines Hot/Cool/Archive storage type. This operation
 // does not update the blob's ETag.
 // For detailed information about block blob level tiering see https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers.
-func (b *BlobClient) SetTier(ctx context.Context, tier AccessTier, options *SetTierOptions) (BlobSetTierResponse, error) {
-	basics, lease, accessConditions := options.pointers()
+func (b *BlobClient) SetTier(ctx context.Context, tier AccessTier, options *BlobSetTierOptions) (BlobSetTierResponse, error) {
+	basics, lease, accessConditions := options.format()
 	resp, err := b.client.SetTier(ctx, tier, basics, lease, accessConditions)
 
-	return resp, handleError(err)
+	return toBlobSetTierResponse(resp), handleError(err)
 }
 
 // GetProperties returns the blob's properties.
 // For more information, see https://docs.microsoft.com/rest/api/storageservices/get-blob-properties.
-func (b *BlobClient) GetProperties(ctx context.Context, options *GetBlobPropertiesOptions) (GetBlobPropertiesResponse, error) {
-	basics, lease, cpk, access := options.pointers()
+func (b *BlobClient) GetProperties(ctx context.Context, options *BlobGetPropertiesOptions) (BlobGetPropertiesResponse, error) {
```

Comment:
Same comment here about breaking changes. Changing the type of options is a breaking change and should be documented in CHANGELOG.md.

We had a ton of changes in other libraries and we're going to miss a few, but only showing one breaking change here is a bad look.